src/components/TOC.vue

```diff
@@ -1,37 +1,45 @@
 <template>
-  <aside class="toc-container u-flex">
+  <aside class="toc-container">
     <h3>{{ toc.title }}</h3>
-    <p>{{ toc.description }}</p>
-    <ul v-if="toc.items.length">
-      <li class="u-flex" v-for="(item, index) in toc.items" :key="index">
-        <span class="ref">{{ index + 1 }}.</span>
-        <div class="item u-flex">
-          <router-link
-            :to="{
-              path: isCiteUrn(item.uri) ? 'tocs' : 'reader',
-              query: { urn: item.uri },
-            }"
-          >
+    <p class="u-legend">{{ toc.description }}</p>
+    <div class="u-grid toc-grid" v-if="toc.items.length">
+      <template v-for="(item, index) in toc.items">
+        <span :key="`index-${index}`" class="ref">{{ index + 1 }}.</span>
+        <div :key="`item-${index}`" class="item u-flex">
+          <router-link :to="getPayload(item.uri)">
             {{ item.title }}
           </router-link>
-          <span>
+          <span v-if="showURNs">
             <tt>{{ item.uri }}</tt>
           </span>
         </div>
-      </li>
-    </ul>
+      </template>
+    </div>
     <h4 v-else>No results.</h4>
   </aside>
 </template>
 
 <script>
   export default {
     name: 'TOC',
-    props: ['toc'],
+    props: ['toc', 'context', 'passage', 'showURNs'],
     methods: {
       isCiteUrn(urn) {
         return urn.startsWith('urn:cite:');
       },
+      getPayload(urn) {
```

Comment:
this is great, and will likely be pretty straightforward to override for the legacy reader to support Brill